zookeeper-client/zookeeper-client-c/src/zookeeper.c

```diff
@@ -3691,6 +3691,21 @@ static int Request_path_watch_init(zhandle_t *zh, int mode,
 /*---------------------------------------------------------------------------*
  * ASYNC API
  *---------------------------------------------------------------------------*/
+#define nonblocking_send(zh, rc) _nonblocking_send(zh, rc, __LINE__)
+static int _nonblocking_send(zhandle_t *zh, int rc, int line)
+{
+    if (rc < 0) {
+	rc = ZMARSHALLINGERROR;
+    } else {
+	rc = adaptor_send_queue(zh, 0);
+	if (rc < 0) {
+	    rc = handle_socket_error_msg(zh, line, rc, "Socket error during nonblocking send");
```

Comment:
I added a commit that addresses this, but it is incidental to the problem of connection loss, so it may be better to put that in a different PR.  Please advise.